sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -146,7 +146,9 @@ abstract class Optimizer(catalogManager: CatalogManager)
         operatorOptimizationRuleSet: _*) ::
       Batch("Push extra predicate through join", fixedPoint,
         PushExtraPredicateThroughJoin,
-        PushDownPredicates) :: Nil
+        PushDownPredicates) ::
+      Batch("Insert window group limit", Once,
+        InsertWindowGroupLimit) :: Nil
```

Comment:
is there a specific reason to put the rule here, instead of the very end of the optimizer? e.g., right before `Batch("User Provided Optimizers", fixedPoint, experimentalMethods.extraOptimizations: _*)` in ·SparkOptimizer·.

Reply:
OK